lib/alchemy/pg_search/engine.rb

```diff
@@ -14,6 +14,8 @@ class Engine < ::Rails::Engine
         # We need to have the search methods present in all Alchemy controllers
         Alchemy::BaseController.send(:include, Alchemy::PgSearch::ControllerMethods)
 
+        # In development environment, this runs on every code reload, so avoid multiple reindexing jobs
+        Alchemy.publish_targets.delete_if { |t| t.name == 'Alchemy::PgSearch::IndexPageJob' }
         # reindex the page after it was published
         Alchemy.publish_targets << Alchemy::PgSearch::IndexPageJob
 
```

Comment:
If would prefer a check if the target is present

```suggestion
        # In development environment, this runs on every code reload, so avoid multiple reindexing jobs
        unless Alchemy.publish_targets.map(&:name).include? 'Alchemy::PgSearch::IndexPageJob'
          # reindex the page after it was published
          Alchemy.publish_targets << Alchemy::PgSearch::IndexPageJob
        end
```